tests/basic/storage_pool_test.go

```diff
@@ -9895,3 +9896,240 @@ var _ = Describe("{AddDriveWithKernelPanic}", func() {
+				}
+			}
+			err = Inst().V.RefreshDriverEndpoints()
+			log.FailOnError(err, "error refreshing volume endpoints")
+		})
+		stepStr = "Test beyond max pool count"
+		Step(stepStr, func() {
+			log.InfoD(stepStr)
+			poolList, err := GetPoolsDetailsOnNode(selectedNode)
+			poolLeft := maxPoolLength - len(poolList)
+			numofDrivesInNode := GetNumOfDrivesInNode(selectedNode)
+			if poolLeft > int(maxDriveLimit)-numofDrivesInNode {
+				stNodes := node.GetStorageNodes()
+				dash.VerifyFatal(len(stNodes) > 0, true, "Storage nodes found?")
+				for _, node := range stNodes {
+					poolList, err := GetPoolsDetailsOnNode(node)
+					log.FailOnError(err, "failed while getting updated node pool list")
+					poolLeft := maxPoolLength - len(poolList)
+					numofDrivesInNode = GetNumOfDrivesInNode(selectedNode)
+					if poolLeft < int(maxDriveLimit)-numofDrivesInNode {
+						selectedNode = node
+						break
+					}
+				}
+			}
+			err = addNewPools(selectedNode, poolLeft)
+			log.FailOnError(err, "error adding new pools")
+			err = addNewPools(selectedNode, 1)
+			dash.VerifyFatal(err != nil, true, "Error expected as pools trying to create is more than allowed")
+		})
+		stepStr = "Test beyond max drives per node"
+		Step(stepStr, func() {
+			log.InfoD(stepStr)
+			var selectedPools string
+			poolList, err := GetPoolsDetailsOnNode(selectedNode)
+			log.FailOnError(err, "failed while getting updated node pool list")
+			//first add all the new pools
+			poolLeft := maxPoolLength - len(poolList)
+			err = addNewPools(selectedNode, poolLeft)
+			//get the number of available drives in the node
+			drvNum, err := Inst().N.RunCommand(selectedNode, "lsblk -l -d -e 11 -n -o NAME|wc -l", node.ConnectionOpts{
+				Timeout:         defaultCommandTimeout,
+				TimeBeforeRetry: defaultCommandRetry,
+				IgnoreError:     false,
+				Sudo:            false,
+			})
+			log.FailOnError(err, "failed to run command")
+			drvNum1 := strings.TrimSpace(string(drvNum))
+			drvNum1 = strings.Trim(drvNum1, "\n")
+			drvNumInt, err := strconv.Atoi(drvNum1)
+			log.FailOnError(err, "failed to convert to int")
+			maxDriveLimitInt := int(maxDriveLimit)
+			maxDriveAllowed := maxDriveLimitInt - drvNumInt
+			err = Inst().V.RefreshDriverEndpoints()
+			log.FailOnError(err, "error refreshing volume endpoints")
+			//expand by adding disk
+			selectedNodeForOps, err := node.GetNodeByName(selectedNode.Name)
+			log.FailOnError(err, "failed while getting node")
+			poolListForOps, err := GetPoolsDetailsOnNode(selectedNodeForOps)
+			log.FailOnError(err, "failed while getting updated node pool list")
+
+			//add max drive allowed per node
+			for maxDriveAllowed > 0 {
+				for i := 0; i < len(poolListForOps); i++ {
+					err = Inst().V.RefreshDriverEndpoints()
+					log.FailOnError(err, "error refreshing volume endpoints")
+					isjournal, err := IsJournalEnabled()
```

Comment:
no need to call this in loop, just call once at start of the test

Reply:
Done